app/Tasks/Types/TheoreticalExam.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Tasks\Types;
+
+use App\Models\Task;
+use App\Models\Training;
+use App\Models\User;
+
+class TheoreticalExam extends Types
+{
+    public function getName()
+    {
+        return 'Theoretical Exam Access';
+    }
+
+    public function getIcon()
+    {
+        return 'fa-key';
+    }
+
+    public function getText(Task $model)
+    {
+        return 'Grant theoretical exam access for ' . Training::find($model->reference_training_id)->getInlineRatings(true);
```

Comment:
note (out of scope): because of the way we handle multiple models with a single inherited model, this is probably a tricky part to avoid additional queries. I think we'd need a proper single inheritance table implementation to avoid that issue.

Reply:
You mean in case of this getText would suddenly ask for some new information that doesn't exist in the Task model or in a lazy loaded relation?

Reply:
Mhm, for each `TheoreticalExam` task, another query will fetch the relevant `Training`. If you've got 10-20 of these tasks, each will need to make a query before it displays. Ideally, the single table needs a way to refer to and eagerly load relevant details to a sub-task type. I've come across a few implementations of this, but I think it's something we can tinker with in a follow-up issue.

Reply:
Yeah that makes sense. Good to note this indeed. We'll look into this later(tm)